File: api/internal/models/command.go
package models

import (
	"encoding/json"
	"time"

	"github.com/google/uuid"
)

type CommandStatus string

const (
	CommandStatusQueued     CommandStatus = "queued"
	CommandStatusInProgress CommandStatus = "in_progress"
	CommandStatusCompleted  CommandStatus = "completed"
	CommandStatusFailed     CommandStatus = "failed"
	CommandStatusExpired    CommandStatus = "expired"
)

type CommandType string

const (
	CommandTypeRefreshNow CommandType = "refresh_now"
)

type Command struct {
	ID          uuid.UUID       `json:"id" db:"id"`
	DeviceID    uuid.UUID       `json:"device_id" db:"device_id"`
	CommandType CommandType     `json:"command_type" db:"command_type" validate:"required"`
	Payload     json.RawMessage `json:"payload" db:"payload"`
	Status      CommandStatus   `json:"status" db:"status"`
	CreatedAt   time.Time       `json:"created_at" db:"created_at"`
	ExecutedAt  *time.Time      `json:"executed_at" db:"executed_at"`
	Result      json.RawMessage `json:"result" db:"result"`
}

// CommandRequest represents a request to create a new command
type CommandRequest struct {
	CommandType CommandType     `json:"command_type" validate:"required"`
	Payload     json.RawMessage `json:"payload"`
}

// CommandResult represents the result of command execution
type CommandResult struct {
	Success bool   `json:"success"`
	Message string `json:"message,omitempty"`
	Error   string `json:"error,omitempty"`
}

// RefreshNowPayload represents the payload for refresh_now commands
type RefreshNowPayload struct {
	Force bool `json:"force,omitempty"`
}

File: web/types/command.ts
// Command status enum
export type CommandStatus = 'queued' | 'in_progress' | 'completed' | 'failed' | 'expired'

// Command type enum
export type CommandType = 'refresh_now'

// Command interface
export interface Command {
  id: string
  device_id: string
  command_type: CommandType
  payload: Record<string, unknown> | null
  status: CommandStatus
  created_at: string
  executed_at?: string
  result: Record<string, unknown> | null
}

// Command creation request
export interface CommandRequest {
  command_type: CommandType
  payload?: Record<string, unknown>
}

// Command execution result
export interface CommandResult {
  success: boolean
  message?: string
  error?: string
}

// Refresh now payload
export interface RefreshNowPayload {
  force?: boolean
}